### DISCUSSION TITLE
(sysinternals) Include ARM64 binaries in package

### DISCUSSION THREAD

**Discussion by cxiao** [chocolatey-community/chocolatey-packages#2062] [Ideas]
On the [official Downloads page for Sysinternals](https://learn.microsoft.com/en-us/sysinternals/downloads/sysinternals-suite), Microsoft currently provides the following versions of Sysinternals for download:

- Sysinternals for x86_32 and x86_64 systems, at https://download.sysinternals.com/files/SysinternalsSuite.zip
- Sysinternals for Nano server, at https://download.sysinternals.com/files/SysinternalsSuite-Nano.zip
- Sysinternals Suite for ARM64, at https://download.sysinternals.com/files/SysinternalsSuite-ARM64.zip

The Sysinternals package on the Chocolatey community repository currently only provides the x86_32 / x86_64 and Nano server versions of Sysinternals.

While a lot of the Sysinternals tools *will* work just fine under Windows ARM64's x86_32 and x86_64 instruction emulation, tools which rely on loading a kernel driver for their main functionality will not. An example of such a tool is Procmon, which cannot provide its main functionality without loading the ARM64 version of the `PROCMON24.sys`driver; the ARM64 version of the driver is only available as a resource inside `procmon64a.exe`, which is only included in the ARM64 download of Sysinternals.

Note that not all of the Sysinternals tools have a binary available in `SysinternalsSuite-ARM64.zip`. As of 2022-11-10, this is the list of binaries inside that ZIP:

```
ADExplorer64a.exe
ADInsight64a.exe
Autologon64a.exe
Autoruns64a.exe
Clockres64a.exe
Coreinfo64a.exe
Dbgview64a.exe
DiskView64a.exe
FindLinks64a.exe
LoadOrd64a.exe
LoadOrdC64a.exe
Procmon64a.exe
PsExec64a.exe
RAMMap64a.exe
RegDelNull64a.exe
Sysmon64a.exe
Winobj64a.exe
ZoomIt64a.exe
accesschk64a.exe
adrestore64a.exe
autorunsc64a.exe
disk2vhd64a.exe
diskext64a.exe
du64a.exe
handle64a.exe
hex2dec64a.exe
junction64a.exe
logonsessions64a.exe
movefile64a.exe
notmyfault64a.exe
notmyfaultc64a.exe
pendmoves64a.exe
pipelist64a.exe
procdump64a.exe
procexp64a.exe
ru64a.exe
sdelete64a.exe
sigcheck64a.exe
streams64a.exe
strings64a.exe
sync64a.exe
tcpvcon64a.exe
tcpview64a.exe
vmmap64a.exe
whois64a.exe
```

(I haven't been able to find any information from Microsoft that says whether they're planning to provide ARM64 builds eventually for all the tools.)

I have access to a Windows 11 ARM64 machine and am willing to submit a PR to help with this.

**Comment by AdmiringWorm** [comment]
I would be okay with a PR being created, but no one could review/test out the changes you made (AFAIK).

So even if you created a PR, it would likely be left hanging until one or more repository maintainers can test your changes.

If you are still okay with that, I can approve and get an issue created based on this discussion

**Reply by cxiao** [reply]
Thanks, I'm fine with that - getting an issue created is a great first step.

**Reply by AdmiringWorm** [reply]
The issue has been created: #2188 

If you can leave a comment in the created issue saying you will pick it up, we can assign you to the issue.

**Reply by cxiao** [reply]
Done, thank you!
